app/src/main/java/com/github/leuludyha/ibdb/presentation/screen/maps/GoogleMapsScreen.kt

```diff
@@ -1,49 +1,235 @@
+        position = CameraPosition.fromLatLngZoom(viewModel.defaultLocation, 15f)
     }
 
+    val uiSettings by remember {
+        mutableStateOf(
+            MapUiSettings(
+                //I did my own implementation of this, I encountered problems and no documentation online
+                myLocationButtonEnabled = false,
+                zoomControlsEnabled = false
+            )
+        )
+    }
+    val context = LocalContext.current
+    val nearbyUsers by viewModel.nearbyUsers
+
     GoogleMap(
         modifier = Modifier
+            .padding(paddingValues)
             .fillMaxSize()
             .testTag("GoogleMaps::main"),
+        uiSettings = uiSettings,
         cameraPositionState = cameraPositionState
     ) {
         EpflMarkerAndLimits()
         SatelliteMarker()
+
+        nearbyUsers.forEach { user ->
+            BookSharingMarker(user = user, context = context)
+        }
+    }
+
+    MapsUI(
+        viewModel = viewModel,
+        paddingValues = paddingValues,
+        cameraPositionState = cameraPositionState,
+        context = context
+    )
+}
+
+/**
+ * This is the UI that will be used on top the map. It consists of a composable that should be superimposed
+ * onto the map. Right now, it contains two buttons for location and refreshing.
+ * @param viewModel of the maps screen
+ * @param paddingValues of the maps screen (make sure they are the same so that it properly is on top of the map!
+ * @param cameraPositionState of the map. It might be updated from calls to the UI
+ * @param context of the map
+ */
+@OptIn(ExperimentalPermissionsApi::class)
+//This permission check is actually done in the code, but for some reason the compiler cries it about it not being done without suppressing it.
+@SuppressLint("MissingPermission")
+@Composable
+private fun MapsUI(
+    viewModel: GoogleMapsScreenViewModel,
+    paddingValues: PaddingValues,
+    cameraPositionState: CameraPositionState,
+    context: Context
+) {
+    //This allows to get the location of the user
+    val fusedLocationClient = LocationServices.getFusedLocationProviderClient(context)
+    val locationPermissionState =
+        rememberPermissionState(permission = Manifest.permission.ACCESS_FINE_LOCATION)
+    val coroutineScope = rememberCoroutineScope()
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(paddingValues),
+        horizontalAlignment = Alignment.End,
+        verticalArrangement = Arrangement.Bottom
+    ) {
+        //This is the refresh users button
+        MapsButton(Modifier.testTag("GoogleMaps::refresh_button"), Icons.Filled.Autorenew) {
+            viewModel.fetchNearbyUsers(cameraPositionState, context).thenAccept {
+                //TODO: Remove the mock users from the list once we can add users to Firebase
+                // for testing and examples
+                viewModel.nearbyUsers.value = it.plus(Mocks.userList)
```

Comment:
The nearby users are not displayed at first before refreshing. It is quite counter intuitive to have to refresh to display them... Could you display them once by default at the beginning?

Reply:
I have done this in a new commit (without adding the mock users to it though, so it doesn't actually fetch anything for the time being). Note that in its current form, maps doesn't focus on the user when opening the map either, just for development reasons. It turned out to be quite challenging changing the location on the emulator and properly making it work, and by default the location of the emulator is at google HQ in Sydney. Therefore I judged it better not to focus on the user nor retrieving nearby users at first, because it greatly complicates using the emulator on android studio. Changing this only requires one line of code though, so its is not a big deal to change it when "shipping the product". Since Firebase now works, there is no harm anymore on fetching the users when opening the map though, so I added it.